### PR TITLE
Support psci calls (system_off only for now) through hvc

### PR DESCRIPTION
**Pull request by aarkegz**

### CODE REVIEW

File: src/exception.rs [hunk 1]
```diff
@@ -103,6 +103,11 @@ pub fn handle_exception_sync(ctx: &mut TrapFrame) -> AxResult<AxVCpuExitReason>
             // The `#imm`` argument when triggering a hvc call, currently not used.
             let _hvc_arg_imm16 = ESR_EL2.read(ESR_EL2::ISS);
 
+            // Is this a psci call?
```
Comment (pengzechen) [inline src/exception.rs:106]: More things need to there.
Guest maybe do smc call to handle PSCI.
In linux, you can give it a dtb.
```
        psci {
		migrate = <0xc4000005>;
		cpu_on = <0xc4000003>;
		cpu_off = <0x84000002>;
		cpu_suspend = <0xc4000001>;
		method = "smc";
		compatible = "arm,psci-1.0\0arm,psci-0.2\0arm,psci";
	};
```

Comment (hky1999) [reply src/exception.rs:106]: Yep, but PSCI related device info should be managed elsewhere.

File: src/exception.rs [hunk 2]
```diff
@@ -103,6 +103,11 @@ pub fn handle_exception_sync(ctx: &mut TrapFrame) -> AxResult<AxVCpuExitReason>
...
+            if let Some(result) = handle_psci_hvc(ctx) {
```
Comment (hky1999) [inline src/exception.rs:107]: Is there any other method for distinguishing HVC and SMC here?

Comment (pengzechen) [reply src/exception.rs:107]:
```C
    else if (ec == 0x16)
    { // hvc
        print_info("            This is hvc call handler\n");
        return;
    }
    else if (ec == 0x17)
    { // smc
        ept_violation_info_t info;
        //printf("Prefetch abort : %x\n", hsr.bits);
        info.hsr.bits = hsr.bits;
        print_info("            This is smc call handler\n");
        advance_pc(&info, ctx_el2);
        return;
    }
```
ESR_EL2::EC::  <=> ec

Comment (hky1999) [reply src/exception.rs:107]: @aarkegz

Comment (aarkegz) [reply src/exception.rs:107]: Yes, but the questions here is that SMC is not trapped now. (`HCR.TSC` is not set, it was but later this line was commented out `src/vcpu.rs:191`, we need to figure out why in the first place.)

Comment (aarkegz) [reply src/exception.rs:107]: Handling HVCs only is a good workaround for the current situation. It's minimal and can be easily extended to a full implementation.

Comment (hky1999) [reply src/exception.rs:107]: > Yes, but the questions here is that SMC is not trapped now. (`HCR.TSC` is not set, it was but later this line was commented out `src/vcpu.rs:191`, we need to figure out why in the first place.)

Yep, I think that we also need to modify arceos's [boot.rs](https://github.com/arceos-hypervisor/arceos/blob/8f55599a4c9caf1c654564668e51b20fc939bc67/modules/axhal/src/platform/aarch64_common/boot.rs#L208)

File: src/exception.rs (hunk 1 above)
Comment (hky1999) [inline src/exception.rs:106]: Add more comments here please.

Expecially for ArceOS, when its platform is qemu-virt-aarch64, it use hvc to trigger `psci_call`

**Review by hky1999 — APPROVED** [review]

File: src/exception.rs [hunk 3]
```diff
@@ -187,6 +192,30 @@ fn handle_data_abort(context_frame: &mut TrapFrame) -> AxResult<AxVCpuExitReason
     })
 }
 
+/// Handles HVC exceptions that serve as psci (Power State Coordination Interface) calls.
+///
+/// Returns `None` if the HVC is not a psci call.
+fn handle_psci_hvc(ctx: &mut TrapFrame) -> Option<AxResult<AxVCpuExitReason>> {
```
Comment (hky1999) [inline src/exception.rs:198]: handle_psci

Both `smc` and `hvc` handlers can call this function

Comment (aarkegz) [reply src/exception.rs:198]: OK

File: src/exception.rs (hunk 1 above)
Comment (aarkegz) [reply src/exception.rs:106]: OK

**Review by pengzechen — APPROVED** [review]

File: src/exception.rs (hunk 1 above)
Comment (aarkegz) [reply src/exception.rs:106]: Added

File: src/exception.rs (hunk 3 above)
Comment (aarkegz) [reply src/exception.rs]: Added

**Review by hky1999 — APPROVED** [review]

**Pull request merged**